src/main/java/com/google/devtools/common/options/OptionsParserImpl.java

```diff
@@ -806,7 +806,13 @@ private ParsedOptionDescriptionOrIgnoredArgs identifyOptionAndPossibleArgument(
       } else {
         suggestion = "";
       }
-      throw new OptionsParsingException("Unrecognized option: " + arg + suggestion, arg);
+
+      if (!isFirstParsing) {
+          throw new OptionsParsingException("Unrecognized option: " + arg + suggestion, arg);
+      } else {
+          swapShorthandAlias(arg);
```

Comment:
My first thought is this wouldn't have any effect?

I'm I'm reading this right, we reach this line on `--foo` before `MODULE.bazel` has mapped `--foo=--//actual`. Wouldn't that mean there isn't yet any alias mapping for `--foo` so this function couldn't do anything?